### ISSUE TITLE
Investigate support for es6 modules

### ISSUE THREAD

**Issue by mike-marcacci** [mike-marcacci/fs-capacitor#17]
I've done a rework of fs-capacitor to support node 13 that involved using the `readable-stream` package instead of node's stream library in order to simply continued compatibility with node v8.x, which we can drop at the end of 2019.

I spent the better part of a day trying to figure out exporting interoperable es6 and commonjs modules now that I have a commonjs dependency, and I finally just gave up. Without any useful guidance or pathway put forward by node for interoperability, the only choice here is to chose a single module system, which will be commonjs for the immediate future.

I'm very unhappy with this situation and fear this is going to severely fragment the ecosystem, but don't have the time to get involved any further.

**Comment by mike-marcacci** [comment]
Preparing and testing only js files succeeds an node 8, 10, and 13:

```bash
if [ -d 'lib' ]; then rm -rf lib; fi; npx babel src -d lib && npx tap --test-ignore=src --100
```

**Comment by jaydenseric** [comment]
I suggest following the approach in `graphql-upload` [v11.0.0](https://github.com/jaydenseric/graphql-upload/releases/tag/v11.0.0):

https://github.com/jaydenseric/graphql-upload/commit/cab7737015f388f6ad01753628562ccc01327a84

It should be plain sailing.

**Comment by jaydenseric** [comment]
In the mean time, I don't think the readme should have ESM `import` syntax in the code examples, as ESM is not supported:

<img width="300" alt="Screen Shot 2020-05-30 at 1 52 34 pm" src="https://user-images.githubusercontent.com/1754873/83318939-f4819980-a27c-11ea-840b-7a9e11f14a00.png">

**Comment by mike-marcacci** [comment]
This was fixed by #44 and will be released as version 7 in May.
